### ISSUE TITLE
Question about experimental comparison

### ISSUE THREAD

**Issue by stevenyangyj** [mbzuai-metaverse/VOODOO3D-official#4]
Hi, great work! I would like to know how you compare VOODOO3D with MegaPortraits which is not open-sourced (see [here](https://github.com/neeek2303/MegaPortraits/issues/1)).

Thanks!

**Comment by johndpope** [comment]
Claude OPUS - response
(just give html from both papers)
<img width="1139" alt="Screenshot 2024-04-20 at 7 39 36 am" src="https://github.com/MBZUAI-Metaverse/VOODOO3D-official/assets/289994/f64c6c46-7b61-4de9-886d-819c6685ca4f">



There are several key similarities and differences between the VOODOO 3D and MegaPortraits papers:

Similarities:

Both aim to create high-quality, one-shot 3D head avatars from a single image. The goal is realistic reenactment of the source identity with the motion/expression of a driving video.
Both use neural network architectures and training techniques to learn volumetric representations and warp fields to map between source appearance and driver motion.
Both demonstrate state-of-the-art results in terms of identity preservation, motion transfer, and photorealism compared to previous methods.
Key Differences:

VOODOO 3D uses a fully volumetric approach, encoding both source and driver into shared tri-plane representations. MegaPortraits encodes source into a 3D volume but uses latent vectors for driver motion.
VOODOO 3D aims to disentangle appearance and motion in a self-supervised way using the shared tri-plane representation and losses. MegaPortraits uses separate encoders and adversarial losses.
VOODOO 3D showcases results up to 512x512 resolution. MegaPortraits focuses on extending to 1024x1024 using a second high-res image dataset.
VOODOO 3D trains end-to-end, while MegaPortraits uses a two-stage approach - first training a 512x512 model, then training an enhancer to 1024x1024.
MegaPortraits demonstrates a distillation technique to create a real-time student model locked to predefined identities. VOODOO 3D does not cover this.
In summary, both push the state-of-the-art in one-shot 3D avatars, but take different approaches - VOODOO 3D focused on self-supervised volumetric disentanglement, while MegaPortraits highlighted scaling to megapixel resolution and real-time distillation. Comparing their approaches on the same dataset could yield further insights.

**Comment by P0lyFish** [comment]
Hi,
We have a reimplementation version of MegaPortraits as we mentioned in the paper.

**Comment by BIGJUN777** [comment]
> Hi, We have a reimplementation version of MegaPortraits as we mentioned in the paper.

@P0lyFish Hi there, we tried to reproduce MegaPortrait but encountered some issues in the part of 3D warping. do you have a plan to share the reimplementation of MegaPortrait or some reference materials?

**Comment by johndpope** [comment]
this may help @BIGJUN777  https://real3dportrait.github.io/

**Comment by BIGJUN777** [comment]
> this may help @BIGJUN777 https://real3dportrait.github.io/

Thanks. I quickly browsed the repo but it seems not helpful because they used a different canonical coordinates space.

**Comment by johndpope** [comment]
@BIGJUN777  did you see this https://github.com/Kevinfringe/MegaPortrait/blob/master/model.py#L82

I have a branch with some fixes https://github.com/johndpope/MegaPortrait/
